front/src/components/login/register.jsx

```diff
@@ -0,0 +1,171 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import axios from "axios";
+import { FormErrors } from "./FormErrors";
+export class Register extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			username: "",
+			password: "",
+			formErrors: {
+				username: "",
+				password: "",
+				email: "",
+				isChecked: false,
+				usernameValid: false,
+				emailValid: false,
+				passwordValid: false,
+				formValid: false
+			}
+		};
+		this.handleUserInput = this.handleUserInput.bind(this);
+		this.handleChecked = this.handleChecked.bind(this);
+	}
+	handleUserInput = e => {
+		const name = e.target.name;
+		const value = e.target.value;
+		this.setState({ [name]: value }, () => {
+			this.validateField(name, value);
+		});
+	};
+	handleChecked(e) {
+      this.setState({ isChecked: !this.state.isChecked });
+      console.log(this.state.isChecked);
+    }
+	validateField(fieldName, value) {
+		let fieldValidationErrors = this.state.formErrors;
+		let usernameValid = this.state.usernameValid;
+		let passwordValid = this.state.passwordValid;
+		let emailValid = this.state.emailValid;
+
+		switch (fieldName) {
+			case "email":
+				emailValid = value.match(
+					/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/
+				);
+				fieldValidationErrors.email = emailValid ? "" : " неправильный ";
+				break;
+			case "username":
+				usernameValid = value.match(/^[a-zA-Z0-9]+$/);
+				fieldValidationErrors.username = usernameValid ? "" : " используйте только a-z A-Z 0-9";
+				break;
+			case "password":
+				passwordValid = value.length >= 6;
+				fieldValidationErrors.password = passwordValid ? "" : "слишком короткий";
+				break;
+			default:
+				break;
+		}
+		this.setState(
+			{
+				formErrors: fieldValidationErrors,
+				usernameValid: usernameValid,
+				passwordValid: passwordValid,
+				emailValid: emailValid
+			},
+			this.validateForm
+		);
+	}
+	validateForm() {
+		this.setState({
+			formValid:
+				this.state.usernameValid &&
+				this.state.passwordValid &&
+				this.state.emailValid
+		});
+	}
+	errorClass(error) {
+		return error.length === 0 ? "" : "has-error";
+	}
+	PostSend() {
+		console.dir(this.setState());
+		axios
+			.post("/User/SignUP", {
+				username: this.state.username,
+				passsword: this.state.password,
+				email: this.state.email
+			})
+			.then(
+				response => {
+					console.log(response);
+					console.log(this.state);
+				},
+				error => {
+					console.log(error);
```

Comment:
Would be cool if you somehow showed the information to the usser instead of console logging, maybe a pop-up error screen/modal?(or if that's hard, just do window.alert(error);

Reply:
I think you're right, it's worth fixing.